client/app/components/ParameterMappingInput.jsx

```diff
@@ -158,6 +159,13 @@ export class ParameterMappingInput extends React.Component {
       newMapping.param = newMapping.param.clone();
       newMapping.param.setValue(newMapping.value);
     }
+    if (has(update, 'type')) {
```

Comment:
Is this an indication that the code is bad? When will there not be a `type`?

Reply:
I think so it is an indication that code is bad, but because it's missing a treatment to the values when the type changes. It's like I mentioned, from my understanding the "value" is coupled to the Static type, and there's no function to handle such change.

There can be no type because I'm checking the updates, not the actual value :slightly_smiling_face:.

Reply:
That's actually similar to changing the Parameter type without updating relevant properties.

Reply:
I misunderstood it. 10x for the clarification 👍